.github/container/Dockerfile.base

```diff
@@ -127,7 +129,11 @@ git apply </opt/pip/pip-vcs-equivalency.patch
 git add -u
 git commit -m 'Adds JAX_TOOLBOX_VCS_EQUIVALENCY as a trigger to treat all github VCS installs for a package as equivalent. The spec of the last encountered version will be used'
 EOF
-RUN pip install --upgrade --no-cache-dir -e /opt/pip pip-tools && rm -rf ~/.cache/*
+# Create a system-wide venv for the pip-installed world inside the containers
+RUN python -m venv --prompt jax /opt/venv && /opt/venv/bin/pip install --ignore-installed --no-cache-dir -e /opt/pip pip-tools
+# Make sure `python` refers to the venv version
+ENV PATH=/opt/venv/bin:${PATH}
```

Comment:
Is this sufficient instead of `source /opt/venv/bin/activate`?

Reply:
Yes, it. The search mechanism starts checking the PATH one at a time till first match. So, even if there is another python in the PATH, it picks up the first one:
```
$> docker run -it --rm ghcr.io/nvidia/jax-toolbox-internal:11967699553-base-amd64 bash -c "which python && which pip"
/opt/venv/bin/python
/opt/venv/bin/pip
```

Reply:
`/opt/venv/bin/activate` performs additional actions in addition to appending to `PATH`. Are we sure it is OK to omit these in our container?

Reply:
> There was some discussion that using [#1122 (comment)](https://github.com/NVIDIA/JAX-Toolbox/pull/1122#issuecomment-2442330737) instead could be lower risk.

IMHO `PIP_BREAK_SYSTEM_PACKAGES=1` is just fine because we are inside a container.

Reply:
The only problem arises is like this:
```
Found existing installation: PyGObject 3.48.2
ERROR: Cannot uninstall 'PyGObject'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
```
So IMHO, venv is better solution to avoid this kind of issues.

Reply:
The "uninstallable" packages come from line 73 if this file (see above)